Sources/WebDriver/HTTPWebDriver.swift

```diff
@@ -3,32 +3,60 @@ import Foundation
 import FoundationNetworking
 #endif
 
+/// Thrown if we fail to detect the protocol a webdriver server uses.
+public struct ProtocolDetectionError: Error {}
+
 /// A connection to a WebDriver server over HTTP.
 public struct HTTPWebDriver: WebDriver {
-    let rootURL: URL
+    private let serverURL: URL
     public let wireProtocol: WireProtocol
 
     public static let defaultRequestTimeout: TimeInterval = 5 // seconds
 
     public init(endpoint: URL, wireProtocol: WireProtocol) {
-        rootURL = endpoint
+        serverURL = endpoint
         self.wireProtocol = wireProtocol
     }
 
+    public static func createWithDetectedProtocol(serverURL: URL) throws -> HTTPWebDriver {
```

Comment:
Can you help me understand why the label on the parameter?

Reply:
To me, the name of the function doesn't clearly hint about what its a parameter would be